libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallback.java

```diff
@@ -0,0 +1,41 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+
+import com.mapbox.services.android.navigation.v5.utils.DownloadTask;
+
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * Triggers the downloading of the tar file included in the {@link ResponseBody} onto disk.
+ */
+class TarFetchedCallback implements Callback<ResponseBody> {
+
+  private final RouteTileDownloader downloader;
+  private final DownloadTask downloadTask;
+
+  TarFetchedCallback(RouteTileDownloader downloader, DownloadTask downloadTask) {
+    this.downloader = downloader;
+    this.downloadTask = downloadTask;
+  }
+
+  @Override
+  public void onResponse(@NonNull Call<ResponseBody> call, @NonNull Response<ResponseBody> response) {
+    if (response.isSuccessful()) {
+      downloadTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, response.body());
+    } else {
+      OfflineError error = new OfflineError(response.message());
```

Comment:
It'd be 💯 appending the response code to the error message.

Reply:
For extra context these are the messages used in iOS https://github.com/mapbox/mapbox-navigation-ios/pull/1892

Reply:
Deep link -> [here](https://github.com/mapbox/mapbox-navigation-ios/pull/1892/files#diff-036fe67124193d8d28ac725df83050dd)